packages/core/src/asciidoctor-core-api.js

```diff
@@ -2129,3 +2129,111 @@ Timings.prototype.printReport = function (to, subject) {
   outputFunction(' Time to convert document: ' + this.$convert().toFixed(2))
   outputFunction(' Total time (read, parse and convert): ' + this.$read_parse_convert().toFixed(2))
 }
+
+/**
+ * @namespace
+ */
+var SyntaxHighlighter = Opal.const_get_qualified(Opal.Asciidoctor, 'SyntaxHighlighter', true)
+
+// Alias
+Opal.Asciidoctor.SyntaxHighlighter = SyntaxHighlighter
+
+/**
+ * Associates the syntax highlighter class or object with the specified names.
+ *
+ * @memberof SyntaxHighlighter
+ * @param syntaxHighlighter - A {SyntaxHighlighter} Class or Object instance
+ * @param {string|Array} names - A {string} name or an {Array} of {string} names
+ */
+SyntaxHighlighter.register = function (syntaxHighlighter, names) {
+  var args = [syntaxHighlighter].concat(names)
+  SyntaxHighlighter.$register.apply(SyntaxHighlighter, args)
+}
+
+/**
+ * Retrieves the syntax highlighter class or object registered for the specified name.
+ *
+ * @memberof SyntaxHighlighter
+ * @param {string} name - The {string} name of the syntax highlighter to retrieve.
+ * @returns the {SyntaxHighlighter} Class or Object instance registered for this name.
+ */
+SyntaxHighlighter.for = function (name) {
+  var result = SyntaxHighlighter.$for(name)
+  return result === Opal.nil ? undefined : result
+}
+
+/**
+ * Resolves the name to a syntax highlighter instance, if found in the registry.
+ *
+ * @memberof SyntaxHighlighter
+ * @param {string} name - The {string} name of the syntax highlighter to create.
+ * @param {string} backend - The {string} name of the backend for which this syntax highlighter is being used (default: 'html5').
+ * @param {Object} opts - a JSON of options providing information about the context in which this syntax highlighter is used:
+ * - doc: The Document for which this syntax highlighter was created.
+ * @returns a {SyntaxHighlighter} instance for the specified name.
+ */
+SyntaxHighlighter.create = function (name, backend, opts) {
+  if (typeof backend === 'undefined') backend = 'html5'
+  if (typeof opts === 'undefined') opts = {}
+  return SyntaxHighlighter.$create(name, backend, opts)
+}
+
+/**
+ * @namespace
+ */
+var SyntaxHighlighterBase = Opal.const_get_qualified(SyntaxHighlighter, 'Base', true)
+
+// Alias
+Opal.Asciidoctor.SyntaxHighlighterBase = SyntaxHighlighterBase
+
+/**
+ * Creates a new {SyntaxHighlighter} instance.
+ *
+ * @memberof SyntaxHighlighterBase
+ * @param {string} name - The {string} name of the syntax highlighter to create.
+ * @param {Function} block - A {Function} with the {SyntaxHighlighter} instance passed as `this`
+ * @returns a {SyntaxHighlighter} instance for the specified name.
+ */
+SyntaxHighlighterBase.create = function (name, block) {
```

Comment:
We already define a `create` on `SyntaxHighlighter` so it might be confusing. The main difference is that `SyntaxHighlighter.create` will create an instance of an existing `SyntaxHighlighter`, where `SyntaxHighlighterBase.create` will create a *new* instance.

Reply:
I think exposing both exposes too much. Really, this is really designed to be an internal API anyway. The integrator should only have to register a syntax highlighter. It's up to the processor to create it and make use of it.

The only API I think we need from this side is `SyntaxHighlighter.for`. That resolves what was previously registered, which can be useful for extending and reregistering.

Reply:
I think we also need an API to register a syntax highlighter. It can be cumbersome to create (and instantiate) a new class in Opal that's why I introduces this `create` function.

But we could mimic the extension registry and do something like:


```js
asciidoctor.SyntaxHighlighter.register('unavailable', function () {
  this.format(function (node, language) {
    return `<pre class="highlight"><code class="language-${language}" data-lang="${language}">${node.getContent()}</code></pre>`
  })
  this.highlighted(() => false)
})
```

Or even:

```js
class UnavailableSyntaxHighlighter {
  format(node, language) {
    return `<pre class="highlight"><code class="language-${language}" data-lang="${language}">${node.getContent()}</code></pre>`
  }

  highlighted() {
    return false
  }
}

asciidoctor.SyntaxHighlighter.register('unavailable', UnavailableSyntaxHighlighter)
```

Reply:
I definitely support that idea. I'd say map it in the most consistent way as you see for Asciidcotor.js users.